src/engine/Server.cpp

```diff
@@ -223,26 +223,28 @@ net::awaitable<std::optional<Server::TimeLimit>>
 Server::verifyUserSubmittedQueryTimeout(
     std::optional<std::string_view> userTimeout, bool accessTokenOk,
     const ad_utility::httpUtils::HttpRequest auto& request, auto& send) const {
-  TimeLimit timeLimit = RuntimeParameters().get<"default-query-timeout">();
+  auto defaultTimeout = RuntimeParameters().get<"default-query-timeout">();
+  using DefaultUnit = decltype(defaultTimeout)::DurationType;
   // TODO<GCC12> Use the monadic operations for std::optional
   if (userTimeout.has_value()) {
     auto timeoutCandidate =
         ad_utility::ParseableDuration<TimeLimit>::fromString(
             userTimeout.value());
-    if (timeoutCandidate > timeLimit && !accessTokenOk) {
+    if (TimeLimit{timeoutCandidate} > DefaultUnit{defaultTimeout} &&
```

Comment:
Don't we have a comparison of two duration values? If not, we should have. Then whenever we want to compare two duration values, we can just use that comparison (instead of casting to a particular unit just to be able to compare).

Reply:
Currently the `ParseableDuration` type, which is the type being used here can only be compared with other types using the same template parameters, but both of them can be converted to their corresponding duration type. The alternative to this would be a lot of boilerplate code because we can't use the spaceship operator for `std::chrono::duration` types yet because libc++ doesn't support it yet.